### PR TITLE
compare objects after cloning via api

### PR DESCRIPTION
**Pull request by pondrejk**
related to #3984 
test results:
```
nosetests tests/foreman/api/test_contentview.py:ContentViewCreateTestCase.test_positive_clone 
.
----------------------------------------------------------------------
Ran 1 test in 7.680s
```
```
nosetests tests/foreman/api/test_hostgroup.py:HostGroupTestCase.test_positive_clone
.
----------------------------------------------------------------------
Ran 1 test in 8.493s
```
```
nosetests tests/foreman/api/test_template.py:ConfigTemplateTestCase.test_positive_clone
.
----------------------------------------------------------------------
Ran 1 test in 10.533s
```

### CODE REVIEW

**Comment by oshtaier** [comment]
Could you please provide few examples? Like what fields are left once you delete unique ones

**Comment by pondrejk** [comment]
@oshtaier sure, pasting some api responses:

.../api/v2/hostgroups/40
```
{

    "content_source_id": null,
    "content_source_name": null,
    "content_view_id": null,
    "content_view_name": null,
    "lifecycle_environment_id": null,
    "lifecycle_environment_name": null,
    "subnet_id": null,
    "subnet_name": null,
    "operatingsystem_id": null,
    "operatingsystem_name": null,
    "domain_id": null,
    "domain_name": null,
    "environment_id": null,
    "environment_name": null,
    "compute_profile_id": null,
    "compute_profile_name": null,
    "ancestry": null,
    "parent_id": null,
    "parent_name": null,
    "puppet_proxy_id": null,
    "puppet_proxy_name": null,
    "puppet_ca_proxy_id": null,
    "puppet_ca_proxy_name": null,
    "ptable_id": null,
    "ptable_name": null,
    "medium_id": null,
    "medium_name": null,
    "architecture_id": null,
    "architecture_name": null,
    "realm_id": null,
    "realm_name": null,
    "created_at": "2017-01-05 10:25:47 +0100",
    "updated_at": "2017-01-05 10:25:47 +0100",
    "id": 40,
    "name": "54yQuqvrOjHu1b",
    "title": "54yQuqvrOjHu1b",
    "template_combinations": [ ],
    "puppetclasses": [ ],
    "config_groups": [ ],
    "parameters": [ ],
    "all_puppetclasses": [ ],
    "locations": [ ],
    "organizations": [ ]

}
```

.../api/v2/config_templates/250
```
{

    "template": "test",
    "locked": false,
    "snippet": false,
    "audit_comment": null,
    "created_at": "2017-01-20 12:08:59 +0100",
    "updated_at": "2017-01-20 12:08:59 +0100",
    "id": 250,
    "name": "qvqkok",
    "template_kind_id": 2,
    "template_kind_name": "PXELinux",
    "template_combinations": [ ],
    "operatingsystems": [ ],
    "os_default_templates": [ ],
    "locations": [ ],
    "organizations": [ ]

}
```

.../katello/api/content_views/1
```
{

    "content_host_count": 0,
    "composite": false,
    "repository_ids": [
        36
    ],
    "component_ids": [ ],
    "default": false,
    "next_version": 2,
    "id": 13,
    "name": "3521",
    "label": "3521",
    "description": null,
    "organization": {
        "name": "ihqjjFO",
        "label": "ihqjjFO",
        "id": 3
    },
    "created_at": "2017-01-03 12:55:56 +0100",
    "updated_at": "2017-01-03 12:55:59 +0100",
    "environments": [
        {
            "id": 2,
            "name": "Library",
            "label": "Library",
            "permissions": {
                "readable": true
            }
        },
        {
            "id": 12,
            "name": "jMjPxbabEn",
            "label": "jMjPxbabEn",
            "permissions": {
                "readable": true
            }
        }
    ],
    "repositories": [
        {
            "id": 36,
            "name": "ZXvZfMmtcU",
            "label": "ZXvZfMmtcU",
            "content_type": "yum"
        }
    ],
    "puppet_modules": [ ],
    "versions": [
        {
            "id": 13,
            "version": "1.0",
            "published": "2017-01-03 12:55:59 +0100",
            "environment_ids": [
                2,
                12
            ]
        }
    ],
    "components": [ ],
    "activation_keys": [
        {
            "id": 23,
            "name": "CrCCALgpFY"
        }
    ],
    "last_published": "2017-01-03 12:55:59 +0100",
    "permissions": {
        "view_content_views": true,
        "edit_content_views": true,
        "destroy_content_views": true,
        "publish_content_views": true,
        "promote_or_remove_content_views": true
    }

}

```

So there's a plenty of it, though lot of None fields, I can set some values in test to make it more fun, but still good to check if None remains None after clone :)

File: tests/foreman/api/test_contentview.py [hunk 1]
```diff
@@ -366,6 +366,30 @@ def test_positive_create_with_description(self):
...
+        cloned_cv = entities.ContentView(id=content_view.copy(
+            data={u'name': gen_string('alpha', gen_integer(3, 30))}
+        )['id']).read_json()
+
+        # remove unique values before comparison
+        cv_origin = content_view.read_json()
+        uniqe_keys = (u'label', u'id', u'name', u'updated_at', u'created_at')
+        for key in uniqe_keys:
+            del cv_origin[key]
+            del cloned_cv[key]
```
Comment (rochacbruno) [inline tests/foreman/api/test_contentview.py]: @renzon I think it is annoying to remove unique keys before comparing, what do you think?

Maybe Nailgun can resolve that internally when performing `__eq__` uniq keys can be removed from comparison and only used with `is`

Comment (renzon) [inline tests/foreman/api/test_contentview.py]: I think we need to think on a solution on nailgun. But we could merge this PR and improve test after some nailgun implementation.

**Pull request merged**